R/create_covariate_summary_plot.R

```diff
@@ -0,0 +1,158 @@
+     
+    } else {
+      stop("The outcome should be either continuous or binary")
+    }
+  }
+  
+  # If there is a covariate 
+  else {
+    
+    # Use FindCovariateName function to find the covariate column
+    covariate <- FindCovariateNames(all_data)[1]
+    
+    # Use GetFriendlyCovariateName function to label the y-axis
+    y_axis_label <- GetFriendlyCovariateName(covariate)
+    
+    caption_setting <- "covariate" # Text to add to caption
+    mutated_data <- all_data # Rename for input to BUGSnet::data.prep
+  } 
+  
+  # Add column with treatment labels
+  mutated_data <- mutated_data %>%
+    dplyr::inner_join(treatment_df, by = join_by(T == Number)) 
+  
+  # Convert tibble created by dplyr to df
+  BUGSnet_df <- as.data.frame(mutated_data)
+    
+  # BUGSnet data prep to convert data to format required for data.plot
+  BUGSnet_data <- BUGSnet::data.prep(arm.data = BUGSnet_df,
+                                     varname.t = 'Label',
+                                     varname.s = 'Study')
+  
+  # Baseline risk plot with error bars
+  if (caption_setting == "baseline risk") {
+    
+    plot <- BUGSnet::data.plot(BUGSnet_data,
+                               covariate = covariate,
+                               covariate.label = y_axis_label,
+                               half.length = "baseline_error",
+                               by = 'treatment',
+                               text.size = 16)
+    
+    # Add caption text under plot
+    plot <- plot +
+      labs(caption = PasteCaptionText(caption_setting, error_bar_text))
+    
+    if (metaoutcome == "Binary") {
+      
+      # Plot in logit scale, label on probability scale
+      plot <- plot +
+        scale_y_continuous(labels = function(x) round(plogis(x), digits = 2))
+    }
+    
+  # Covariate plot without error bars
+  } else {
+    
+    plot <- BUGSnet::data.plot(BUGSnet_data,
+                               covariate = covariate,
+                               covariate.label = y_axis_label,
+                               by = 'treatment',
+                               text.size = 16) 
+    
+    # Add caption text under plot
+    plot <- plot +
+      labs(caption = PasteCaptionText(caption_setting))
+  }
+  
+  plot <- plot +
+    # Right aligned caption text
+    theme(plot.caption = element_text(hjust = 1)) 
+
+  
+  return(plot)
+  
+}
+
+#' Paste the caption text
+#'
+#' @param plot_type Text string to describe type of plot. Can be "baseline risk" or "covariate"
+#' @param error_bar_text Text string to explain the error bar (optional)
+#' @return Text string to be used for caption
+
+PasteCaptionText <- function(caption_setting, error_bar_text = NULL) {
+  
+  # Short lines because the line lengths are not reactive to plot width and I haven't found a fix
+  caption_text <- paste('The plotted', caption_setting, 'value is the same for all treatment arms across a study 
```

Comment:
Some of this text is being dropped of the edge of the plot area because the line is too long, or the plot is too short. Also there does seem to be a lot of dead space in the page.
![image](https://github.com/CRSU-Apps/MetaInsight/assets/136462397/00184182-43f4-4a95-9434-973519a128da)
Since we've now dropped the requirement for a sensitivity analysis, there is a lot more real estate on offer.

Reply:
I've just seen that this is a known issue, so Is this issue explicit in another task?

Reply:
I will investigate this further when the plot is rotated as a workaround now may break after that. (Edited the BUGSnet Jira task)